Packs/jamf/Integrations/jamf_v2/jamf_v2.py

```diff
@@ -0,0 +1,982 @@
+        Args:
+            mobile_id: The computer id.
+            lost_message: The lost message.
+        Returns:
+            Mobile device lost response from API.
+        """
+
+        uri = '/mobiledevicecommands/command/DeviceLock'
+        request_body = '<?xml version="1.0" encoding="UTF-8"?>' + \
+                       '<mobile_device_command>' + \
+                       '<general>' + \
+                       '<command>EnableLostMode</command>' + \
+                       f'<lost_mode_message>{lost_message}</lost_mode_message>' + \
+                       '</general>' + \
+                       '<mobile_devices>' + \
+                       '<mobile_device>' + \
+                       f'<id>{mobile_id}</id>' + \
+                       '</mobile_device>' + \
+                       '</mobile_devices>' + \
+                       '</mobile_device_command>'
+
+        res = self._http_request(method='POST', data=request_body, url_suffix=uri, headers=POST_HEADERS,
+                                 resp_type='response', error_handler=self._mobile_lost_erase_error_handler)
+
+        raw_action = json.loads(xml2json(res.content))
+        return raw_action
+
+    def mobile_device_erase_request(self, mobile_id: str = None, preserve_data_plan: bool = False,
+                                    clear_activation_code: bool = False):
+        """Erase mobile device.
+        Args:
+            mobile_id: The computer id.
+            preserve_data_plan: Retain cellular data plans.
+            clear_activation_code: Clear Activation Lock on the device.
+        Returns:
+           Erase mobile device response from API.
+        """
+
+        uri = '/mobiledevicecommands/command/EraseDevice'
+        request_body = '<?xml version="1.0" encoding="UTF-8"?>' + \
+                       '<mobile_device_command>' + \
+                       '<general>' + \
+                       '<command>EraseDevice</command>' + \
+                       f'<preserve_data_plan>{preserve_data_plan}</preserve_data_plan>' + \
+                       f'<clear_activation_code>{clear_activation_code}</clear_activation_code>' + \
+                       '</general>' + \
+                       '<mobile_devices>' + \
+                       '<mobile_device>' + \
+                       f'<id>{mobile_id}</id>' + \
+                       '</mobile_device>' + \
+                       '</mobile_devices>' + \
+                       '</mobile_device_command>'
+
+        res = self._http_request(method='POST', data=request_body, url_suffix=uri, headers=POST_HEADERS,
+                                 resp_type='response', error_handler=self._mobile_lost_erase_error_handler)
+
+        raw_action = json.loads(xml2json(res.content))
+        return raw_action
+
+
+''' HELPER FUNCTIONS '''
+
+
+def pagination(response, limit, page):
+    if limit > MAX_PAGE_SIZE:
+        limit = MAX_PAGE_SIZE
+    return response[page * limit:(page + 1) * limit]
```

Comment:
This should be split into several variables with readable names, it's too complex and bug-prone